### DISCUSSION TITLE
Does the Android SDK maintain a single web socket connection?

### DISCUSSION THREAD

**Discussion by jeff-huston** [GetStream/stream-chat-android#4211] [General]
I'm trying to understand the web socket usage of the Android app. I've seen [this page](https://getstream.io/chat/docs/android/init_and_users/) of the docs. This means calling `connectUser` will open a web socket and increment the concurrent users.

After that point, am I correct that any other calls we make into the Stream SDK will use that same web socket, and won't open any additional ones?

Thanks in advance for the clarification.

**Comment by JcMinarro** [comment]
The StreamChat SDk Will keep only one WebSocket connection opened at the same time.
If there is any error with the opened WS Connection, it will be closed and if the error is not an unrecoverable error we will try to open a new one. On your side you wouldn't need to do anything, whatever EventObserver/Viewmodel/ChatView you were using will use the new WS connection whenever it  starts to be ready
